### PR TITLE
[FLOC-3188] Make remote_logs.log JSON

### PR DESCRIPTION
**Pull request by exarkun**
Fixes https://clusterhq.atlassian.net/browse/FLOC-3188

[Review on Reviewable](https://reviewable.io/reviews/clusterhq/flocker/2020)

### CODE REVIEW

File: admin/acceptance.py [hunk 1]
```diff
@@ -896,9 +898,41 @@ def capture_journal(reactor, host, output_file):
...
+                    message = dict(message=raw_message)
+
+                message[u"_HOSTNAME"] = accumulated.get(
+                    b"_HOSTNAME", b"<no hostname>"
+                )
+                message[u"_SYSTEMD_UNIT"] = accumulated.get(
+                    b"_SYSTEMD_UNIT", b"<no unit>"
+                )
+                output_file.write(json.dumps(message) + b"\n")
+                accumulated.clear()
```
Comment (jml) [inline admin/acceptance.py]: :grey_question: Not necessary, but I'd probably write this with a yield, for a clearer separation of IO & parsing.

**Comment by jml** [comment]
I think this also fixes https://clusterhq.atlassian.net/browse/FLOC-2097

I guess ideally this should be a thing that's in eliot, since extracting mangled logs from systemd is a general problem. However, happy to start with this.

**Comment by jml** [comment]
Thanks for doing this. I've wanted to do this pretty much since I started at ClusterHQ (but fell over at the first hurdle). 

I've made one optional suggestion. Feel free to merge without further review whether or not you do it.

File: admin/acceptance.py (hunk 1 above)
Comment (exarkun) [inline admin/acceptance.py]: Agree that'd be better factoring.  I'll leave a note but I don't think putting this off 'til later will hurt anything.

**Pull request merged**